sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala

```diff
@@ -503,9 +503,10 @@ private[hive] object HiveContext {
   val HIVE_METASTORE_VERSION: String = "spark.sql.hive.metastore.version"
   val HIVE_METASTORE_JARS: String = "spark.sql.hive.metastore.jars"
 
+  private val tempDir = Utils.createTempDir()
+
   /** Constructs a configuration for hive, where the metastore is located in a temp directory. */
   def newTemporaryConfiguration(): Map[String, String] = {
```

Comment:
Its kind of weird for `newTemporaryConfiguration` to not actually return a new config.  Why not just have a lazy val that is used for the execution conf, and reuse that every time we call reset?